core/src/main/java/io/jenkins/pluginhealth/scoring/probes/NumberOfOpenIssuesProbe.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2023 Jenkins Infra
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.kohsuke.github.GHRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.annotation.Order;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * This probe counts the total number of open issues in GitHub and JIRA
+ */
+@Component
+@Order(value = NumberOfOpenIssuesProbe.ORDER)
+public class NumberOfOpenIssuesProbe extends Probe {
+    public static final String KEY = "open-issue";
+    public static final int ORDER = UpdateCenterPluginPublicationProbe.ORDER + 100;
+    public static final String JIRA_HOST = "https://issues.jenkins.io/";
+    public static final String JIRA_API_PATH = "rest/api/latest/search";
+    private static final Logger LOGGER = LoggerFactory.getLogger(NumberOfOpenIssuesProbe.class);
+
+    @Override
+    protected ProbeResult doApply(Plugin plugin, ProbeContext context) {
+        List<String> issueTracker = getIssueTrackerType(context);
+
+        if (issueTracker.size() > 1) {
+            return ProbeResult.success(key(),
+                getJiraIssues(context, plugin.getScm(), plugin.getName()).message().concat(" ") +
+                getGitHubIssues(context, plugin.getScm(), plugin.getName()).message());
```

Comment:
You are concatenating two `ProbeResult`. How does that work?

Reply:
When there are open issues in both `JIRA` and `GitHub` both,  both methods will return the count of open issues. 

Their message is then combined and returned to the user. 

I tried using a switch case and also played around with lambda. But it did not work.

Reply:
you are combining the `toString()` of both `ProbeResult`. If that method is changed to not only returns the content of the `message` field, this won't work anymore. 

Moreover, what happens if `getJiraIssues` is returning an error?

Reply:
Yes. I was thinking the same. It will return a failure message. 

I want to use these two methods. And resolve the case when there are open issues in both the tracker type.